toolchain/base/relational_value_store.h

```diff
@@ -45,6 +45,7 @@ class RelationalValueStore {
   // Given the related ID and a value, stores the value and returns a mapped ID
   // to reference it in the store.
   auto Add(RelatedIdType related_id, ValueType value) -> IdT {
+    CARBON_DCHECK(related_id.index >= 0, "{0}", related_id);
```

Comment:
Why are these CHECKs on the value of the tagged index, instead of the raw index? Shouldn't the value of the tagged index be an implementation detail of tagging?

Reply:
Done in https://github.com/carbon-language/carbon-lang/pull/6310